modules/media/jsx/mediaIndex.js

```diff
@@ -1,121 +1,185 @@
 import React, {Component} from 'react';
+import PropTypes from 'prop-types';
 import Loader from 'Loader';
-import FilterForm from 'FilterForm';
+import FilterableDataTable from 'FilterableDataTable';
 import {Tabs, TabPane} from 'Tabs';
 
 import MediaUploadForm from './uploadForm';
-import formatColumn from './columnFormatter';
 
 class MediaIndex extends Component {
   constructor(props) {
     super(props);
-    loris.hiddenHeaders = ['Cand ID', 'Session ID', 'Hide File', 'File Type'];
 
     this.state = {
+      data: {},
+      error: null,
       isLoaded: false,
-      filter: {},
     };
 
-    // Bind component instance to custom methods
     this.fetchData = this.fetchData.bind(this);
-    this.updateFilter = this.updateFilter.bind(this);
-    this.resetFilters = this.resetFilters.bind(this);
+    this.formatColumn = this.formatColumn.bind(this);
   }
 
   componentDidMount() {
-    this.fetchData();
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
   }
 
   /**
    * Retrieve data from the provided URL and save it in state
    * Additionally add hiddenHeaders to global loris variable
    * for easy access by columnFormatter.
+   *
+   * @return {object}
    */
   fetchData() {
-    $.ajax(this.props.DataURL, {
-      method: 'GET',
-      dataType: 'json',
-      success: function(data) {
-        this.setState({
-          Data: data,
-          isLoaded: true,
-        });
-      }.bind(this),
-      error: function(error) {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data}))
+      .catch((error) => {
+        this.setState({error: error.error});
         console.error(error);
-      },
-    });
+      });
   }
 
-  updateFilter(filter) {
-    this.setState({filter});
-  }
+  /**
+   * Modify behaviour of specified column cells in the Data Table component
+   *
+   * @param {string} column - column name
+   * @param {string} cell - cell content
+   * @param {object} row
+   *
+   * @return {*} a formated table cell for a given column
+   */
+  formatColumn(column, cell, row) {
```

Comment:
This is just a comment, an objective one: I prefer when functions have a single return statement.

Reply:
Okay, I will look into making that happen!